addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2438,6 +2444,50 @@ var SnippetsMenu = Widget.extend({
             }],
         }).open();
     },
+    /**
+     * @private
+     */
+    _onRenameBtnClick: function (ev) {
+        const $snippet = $(ev.target).closest('.oe_snippet');
+        const snippetName = $snippet.attr('name');
+        const confirmText = _t('Confirm');
+        const cancelText = _t('Cancel');
+        const $input = $(`
+            <we-input class="o_we_user_value_widget w-100 mx-1">
+                <div>
+                    <input type="text" autocomplete="chrome-off" value="${snippetName}" class="text-left"/>
+                    <we-button class="o_confirm_btn fa fa-check" title="${confirmText}"/>
+                    <we-button class="o_cancel_btn fa fa-times" title="${cancelText}"/>
+                </div>
+            </we-input>
+        `);
+        $snippet.find('we-button').remove();
+        $snippet.find('span.oe_snippet_thumbnail_title').replaceWith($input);
+        setTimeout(() => {
```

Comment:
Is this `setTimeout` really needed? I don't think to have any issue removing it on my side